### PR TITLE
Add zip importer

### PR DESCRIPTION
**Pull request by ZetaTwo**
Adds a zip importer. It is very similar to the targz importer but it adds an optional option to treat files with other extensions as zip files. This is useful if importing for example Java .jar files or Python .whl/.egg files which are just zip files under the hood.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -220,6 +220,16 @@ This is very similar to the TarGz importer except that it looks for `.rpm` packa
 
 1. `-rpm_repo_path` which should point to the path on the local file system that contains `.rpm` files
 
+#### Zip (and )
```
Comment (meeehow) [inline README.md:223]: You might have forgotten something here :)

File: importers/zip/zip.go [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+}
+
+// RepoPath returns repository path.
+func (r *Repo) RepoPath() string {
+	return r.location
+}
+
+// DiscoverRepo traverses the repository and looks for files that are related to zip base Archives.
+func (r *Repo) DiscoverRepo() ([]hashr.Source, error) {
+
```
Comment (meeehow) [inline importers/zip/zip.go:240]: Maybe delete the extra space here

File: importers/zip/zip_test.go [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+	if !cmp.Equal(wantExtractions, gotExtractions, cmp.AllowUnexported(extraction{})) {
+		t.Errorf("Preprocess() unexpected diff (-want/+got):\n%s", cmp.Diff(wantExtractions, gotExtractions, cmp.AllowUnexported(extraction{})))
+	}
+}
+
+func TestImageFunctions(t *testing.T) {
+	id := "ubuntu-desktop.zip"
+	repoPath := "/tmp/glinux-repo"
+	localzipPath := "/tmp/glinux-repo/20200108.00.00-ubuntu-desktop.zip"
+	remotePath := "/x20/glinux-repo/20200108.00.00-ubuntu-desktop.zip"
```
Comment (meeehow) [inline importers/zip/zip_test.go:398]: Please change the gLinux/x20 paths

**Review by meeehow — CHANGES_REQUESTED** [review]
Looks good, three small nits

**Comment by ZetaTwo** [comment]
Thanks for the review. Fixed.

**Pull request merged**